runtime/serde/serde-form-url/common/src/aws/smithy/kotlin/runtime/serde/formurl/FormUrlSerializer.kt

```diff
@@ -333,6 +333,9 @@ private class FormUrlMapSerializer(
         value.serialize(FormUrlSerializer(buffer, nestedPrefix))
     }
 
+    override fun entry(key: String, value: Document?) =
+        throw SerializationException("document values not supported by form-url serializer")
```

Comment:
Is this lack of support by design, or will it be implemented later?

Reply:
None of the AWS protocols _except_ JSON-based ones support the document type at this time. The Smithy spec for [restXml](https://smithy.io/2.0/aws/protocols/aws-restxml-protocol.html#supported-traits), [awsQuery](https://smithy.io/2.0/aws/protocols/aws-query-protocol.html), etc., explicitly state document is not supported. If they ever add support then we'd need to implement it here and a few other places as well.